src/main/kotlin/ee/carlrobert/codegpt/actions/TrackableAction.kt

```diff
@@ -0,0 +1,37 @@
+package ee.carlrobert.codegpt.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.editor.Editor
+import ee.carlrobert.codegpt.telemetry.TelemetryAction
+import javax.swing.Icon
+
+abstract class TrackableAction(
+  @JvmField protected val editor: Editor,
+  text: String?,
+  description: String?,
+  icon: Icon?,
+  private val actionType: ActionType
+) : AnAction(text, description, icon) {
+
+  abstract fun handleAction(e: AnActionEvent)
+
+  override fun actionPerformed(e: AnActionEvent) {
+    try {
+      handleAction(e)
+      TelemetryAction.IDE_ACTION
+        .createActionMessage()
+        .property("group", null)
+        .property("action", actionType.name)
+        .send()
```

Comment:
Logic changed: Success or Failure telemetry, not both (therefore properties added to error too).
But I can revert if both are needed in case of an error 😅